hudi-common/src/main/java/org/apache/hudi/metadata/FileSystemBackedTableMetadata.java

```diff
@@ -64,6 +111,6 @@ public FileSystemBackedTableMetadata(SerializableConfiguration conf, String data
 
   @Override
   public boolean isInSync() {
-    throw new UnsupportedOperationException();
+    return false;
```

Comment:
this should be `true`? FileSystem and Data timeline cannot be divergent right

Reply:
I guess. But setting it to true breaks https://github.com/apache/hudi/blob/master/hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/metadata/TestHoodieBackedMetadata.java#L456 and others. I guess will look at the tests again to see if those need to be fixed instead. I don't really see `isInSync()` being used anywhere (in source code, apart from tests) except for being stored in stats somewhere. Wonder if its really needed.

Reply:
the stats are important, so let's keep it there. I think this happens because some parts of the tests run with metadata enabled and this is returned as the fallback. If its not too many, can we fix the tests. it does not make sense to have this return `false` semantically.

Reply:
Will do.